### PR TITLE
Switch to `Generator` for random number generation in `dask.array`

### PR DESCRIPTION
**Pull request by erayaslan**
Use ``Generator`` internally instead of ``RandomState`` for random number generation. ``Generator`` based random number generation was introduced via e2c74 and brings improvements compared to now-maintenance-only `RandomState` class. Basically we
- use `da.random.default_rng()` to create a `Generator` based random number generator
- make sure we use the correct distribution names for the new `Generator` class (`randint` -> `integers` and so on)

See https://numpy.org/doc/stable/reference/random/index.html for further details.

- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by erayaslan** [comment]
cc @rjzamora @jakirkham  Switch to `Generator` based random number generation internally in `dask.array`. Basically, I'd like to
- switch to `Generator` internally
- improve docs
- be an example of how to use `Generator` based RNG for our users

Thank you

**Comment by jakirkham** [comment]
cc @pentschev (in case you have thoughts on the CuPy-side)

**Review by pentschev — COMMENTED** [review]
Looks sensible to me. Does it make sense to update the CuPy tests as well or is there a reason not to do that now? They should all be in `dask/array/tests/test_cupy_*.py`.

**Comment by erayaslan** [comment]
> Looks sensible to me. Does it make sense to update the CuPy tests as well or is there a reason not to do that now? They should all be in `dask/array/tests/test_cupy_*.py`.

It does make sense. I have updated `CuPy` tests as well. Thank you

**Review by pentschev — APPROVED** [review]
LGTM, thanks so much for making all these changes @erayaslan !

**Comment by erayaslan** [comment]
@jakirkham any comments? I think we are good to merge

**Comment by erayaslan** [comment]
ping @jakirkham @rjzamora  these looks good to me for merging. Thank you

File: dask/array/core.py [hunk 1]
```diff
@@ -759,16 +759,16 @@ def map_blocks(
     the example below, ``func`` will result in an ``IndexError`` when computing
     ``meta``:
 
-    >>> da.map_blocks(lambda x: x[2], da.random.random(5), meta=np.array(()))
+    >>> da.map_blocks(lambda x: x[2], da.random.default_rng().random(5), meta=np.array(()))
```
Comment (rjzamora) [inline dask/array/core.py:762]:
```suggestion
    >>> rng = da.random.default_rng()
    >>> da.map_blocks(lambda x: x[2], rng.random(5), meta=np.array(()))
```
Not a blocker. Just personally find it clearer to define the generator separately in examples.

File: dask/array/core.py [hunk 2]
```diff
@@ -3458,7 +3458,7 @@ def from_array(
 
     >>> import numpy as np
     >>> import dask.array as da
-    >>> x = np.random.random((100, 6))
+    >>> x = np.random.default_rng().random((100, 6))
```
Comment (rjzamora) [inline dask/array/core.py:3461]:
```suggestion
    >>> rng = np.random.default_rng()
    >>> x = rng.random((100, 6))
```
Same as my other non-blocking suggestion.

**Review by rjzamora — APPROVED** [review]
Thanks for the great work here @erayaslan ! Sorry for letting this sit for so long. I'll make sure these changes get merged as soon as I'm confident CI is green.

**Comment by charlesbluca** [comment]
rerun tests

**Comment by rjzamora** [comment]
Thanks again for the work here @erayaslan !

**Pull request merged**